lab_3/main.py

```diff
@@ -213,16 +324,43 @@ def get_top_k_n_grams(self, k: int, trie_level: int) -> tuple:
             (3, 4), (4, 1), (1, 5), (5, 2), (2, 1)
         )
         """
-        pass
-
+        if not isinstance(k, int) \
+                or not isinstance(trie_level, int)\
+                or k <= 0:
+            return ()
+        for trie in self.tries:
+            if trie.size == trie_level:
+                freq = trie.n_gram_frequencies
+                top = sorted(freq, key=freq.get, reverse=True)
+                top = top[:k]
+                return tuple(top)
+        return ()
     # 8
     def save(self, name: str) -> int:
         """
         Saves language profile into json file
         :param name: name of the json file with .json format
         :return: 0 if profile saves, 1 if any errors occurred
         """
-        pass
+        if not isinstance(name, str):
+            return 1
+        freq_tuple = {}
+        freq = {}
+        for trie in self.tries:
+            freq_tuple |= trie.n_gram_frequencies
```

Comment:
What does it do?

Reply:
unites all n_gram_frequencies in one dict for simplifying the next loop